importers/src/main/groovy/whelk/reindexer/CardRefresher.groovy

```diff
@@ -0,0 +1,63 @@
+package whelk.reindexer
+
+import whelk.Document
+import whelk.Whelk
+import whelk.util.ThreadPool
+import groovy.util.logging.Log4j2 as Log
+
+import java.time.Instant
+
+@Log
+class CardRefresher {
+    static final int BATCH_SIZE = 1000
+
+    Whelk whelk
+    Instant timestamp = Instant.now()
+
+    CardRefresher(Whelk whelk) {
+        this.whelk = whelk
+    }
+
+    void refresh(String collection = null) {
+        try {
+            int counter = 0
+            long startTime = System.currentTimeMillis()
+
+            ThreadPool threadPool = new ThreadPool(whelk.storage.getPoolSize())
+
+            List<Document> documents = new ArrayList<>(BATCH_SIZE)
+            for (document in whelk.storage.loadAll(collection)) {
+                documents.add(document)
+                counter++
+                if (counter % BATCH_SIZE == 0) {
+                    int docsPerSec = (int) ((double) counter) / ((double) ((System.currentTimeMillis() - startTime) / 1000))
+                    log.info("Refreshing $docsPerSec cards per second (running average since process start). Total count: $counter.")
+                    threadPool.executeOnThread(documents, new BatchHandler())
+                    documents = []
```

Comment:
If there is a real performance gain by using initial capacity we probably want to use it here as well (as on line 28)?

Reply:
:rofl: 
probably microscopic impact but good manners since size is known.

Reply:
I mainly bet on uniformity as well. ;)